app/templates/static/form_handler.js

```diff
@@ -137,9 +142,17 @@ document.getElementById('url-form').addEventListener('submit', async function(ev
                         const x = (parseFloat(target.getAttribute('data-x')) || 0) + event.dx;
                         const y = (parseFloat(target.getAttribute('data-y')) || 0) + event.dy;
 
-                        target.style.transform = `translate(${x}px, ${y}px)`;
-                        target.setAttribute('data-x', x);
-                        target.setAttribute('data-y', y);
+                        // Ensure the text stays within the bounds of the image wrapper
+                        const container = target.closest('.image-wrapper');
+                        const containerRect = container.getBoundingClientRect();
+                        const targetRect = target.getBoundingClientRect();
+
+                        const newX = Math.min(Math.max(x, 0), containerRect.width - targetRect.width);
+                        const newY = Math.min(Math.max(y, 0), containerRect.height - targetRect.height);
+
+                        target.style.transform = `translate(${newX}px, ${newY}px)`;
+                        target.setAttribute('data-x', newX);
+                        target.setAttribute('data-y', newY);
```

Comment:
You didn't remove this. Remove it.

Reply:
Thank you for pointing that out, @lgaleana.

I apologize for the oversight. I'll remove the redundant code as you suggested and push the changes shortly.

Reply:
Remove the lines below this comment: "// Ensure the text stays within the bounds of the image wrapper"

Reply:
Thank you for pointing that out, @lgaleana.

I apologize for the oversight. I'll remove the redundant lines below the comment to keep the code clean and focused on the new logic.

I'll make the necessary adjustments and push the changes shortly.